test/src/main/java/org/apache/accumulo/test/UpgradeRelativePathsIT.java

```diff
@@ -0,0 +1,171 @@
+    }
+  }
+
+  // copied from VolumeIT
+  private void createRelativePaths(AccumuloClient client, String tableName, boolean dottedType)
+      throws Exception {
+    TableId tableId = TableId.of(client.tableOperations().tableIdMap().get(tableName));
+
+    SortedSet<Text> partitions = new TreeSet<>();
+    // with some splits
+    for (String s : "c,g,k,p,s,v".split(","))
+      partitions.add(new Text(s));
+
+    client.tableOperations().addSplits(tableName, partitions);
+
+    BatchWriter bw = client.createBatchWriter(tableName);
+
+    // create two files in each tablet
+    for (String s : VolumeChooserIT.alpha_rows) {
+      Mutation m = new Mutation(s);
+      m.put("cf1", "cq1", "1");
+      bw.addMutation(m);
+    }
+
+    bw.flush();
+    client.tableOperations().flush(tableName, null, null, true);
+
+    for (String s : VolumeChooserIT.alpha_rows) {
+      Mutation m = new Mutation(s);
+      m.put("cf1", "cq1", "2");
+      bw.addMutation(m);
+    }
+
+    bw.close();
+    client.tableOperations().flush(tableName, null, null, true);
+
+    client.tableOperations().offline(tableName, true);
+
+    client.securityOperations().grantTablePermission("root", MetadataTable.NAME,
+        TablePermission.WRITE);
+
+    try (Scanner metaScanner = client.createScanner(MetadataTable.NAME, Authorizations.EMPTY)) {
+      metaScanner.fetchColumnFamily(MetadataSchema.TabletsSection.DataFileColumnFamily.NAME);
+      metaScanner.setRange(new KeyExtent(tableId, null, null).toMetadataRange());
+
+      try (BatchWriter mbw = client.createBatchWriter(MetadataTable.NAME)) {
+        for (Map.Entry<Key,Value> entry : metaScanner) {
+          String cq = entry.getKey().getColumnQualifier().toString();
+          if (cq.startsWith(v1.toString())) {
+            Path path = new Path(cq);
+            String relPath;
+            if (dottedType)
+              relPath = "../" + tableId.canonical() + "/" + path.getParent().getName() + "/"
+                  + path.getName();
+            else
+              relPath = "/" + path.getParent().getName() + "/" + path.getName();
+
+            Mutation fileMut = new Mutation(entry.getKey().getRow());
+            fileMut.putDelete(entry.getKey().getColumnFamily(),
+                entry.getKey().getColumnQualifier());
+            fileMut.put(entry.getKey().getColumnFamily().toString(), relPath,
+                entry.getValue().toString());
+            mbw.addMutation(fileMut);
+          }
+        }
+      }
+    }
+  }
+
+  private void verifyPaths(Scanner metaScanner, int depth) {
+    metaScanner.fetchColumnFamily(MetadataSchema.TabletsSection.DataFileColumnFamily.NAME);
+    for (Map.Entry<Key,Value> entry : metaScanner) {
+      String cq = entry.getKey().getColumnQualifier().toString();
+      Path path = new Path(cq);
+      assertTrue("relative path not replaced " + path, path.depth() > depth);
+    }
+    metaScanner.close();
```

Comment:
It's a bit weird to close the closeable here. Should probably wrap calls to verifyPaths with try-with-resources... it'd be easier to read and to reason about, if there's a failure.